src/hatti/views/table.cljs

```diff
@@ -82,21 +83,47 @@
   (let [clj-value (js->clj value :keywordize-keys true)]
     (forms/format-answer field clj-value language true)))
 
+(defn button-formatter
+  [row cell value columnDef dataContext]
+  (let [{:keys [owner project formid]} (:dataset-info @shared/app-state)
+        form-owner (last-url-param owner)
+        project-id (last-url-param project)
+        edit-link (str "/"
```

Comment:
oh boo, might be time to pull the utils out of hatti and milia (and any extra stuff in zebra) and make a clj-utils library that's used and uniform across hatti, milia, and zebra